components/programs/ClassInstanceData.tsx

```diff
@@ -148,6 +155,11 @@ export interface ClassDates {
   demoDay: number;
 }
 
+export interface ClassSDRDates {
```

Comment:
This could be defined in terms of removing the `demoDay` property requirement from `ClassDates`, e.g.

```typescript
type ClassSDRDates = Omit<ClassDates, "demoDay">;
```

Although if this was done from scratch it would probably make more sense to go the other way around and define them in terms of each other, probably with a more general name for the interface, such as:

```typescript
// Naming is hard, it might make more sense to call this one ClassDates
// and have another one for the classes with demo days, e.g. ProgrammingClassDates,
// but that would require more changes to other places
export interface BaseClassDates {
  registrationDeadline: number;
  interview: number;
}

export interface ClassDates extends BaseClassDates {
  demoDay: number;
}
```